usecases/decision_usecase.go

```diff
@@ -219,10 +219,17 @@ func (usecase *DecisionUsecase) CreateDecision(
 		return models.DecisionWithRuleExecutions{}, err
 	}
 
+	pivotsMeta, err := usecase.dataModelRepository.ListPivots(ctx, exec, input.OrganizationId, nil)
+	if err != nil {
+		return models.DecisionWithRuleExecutions{}, err
+	}
+	pivot := models.FindPivot(pivotsMeta, input.TriggerObjectTable, dataModel)
```

Comment:
I would probably go for a solution like :
```go
triggerTable := dataModel.GetTableByName(input.TriggerObjectTable) //To implement
usecase.dataModelRepository.GetPivot(triggerTable.ID)
```
 but both solution work

Reply:
Ok let me see.
`usecase.dataModelRepository.GetPivot(triggerTable.ID)` kind of already exists (ListPivots takes a last pointer tableId argument to filter on a specific table). It still returns a slice of pivots.
`dataModel.GetTableByName(` also exists, it's basically `dataModel.Tables[name]`. 
The advantage of my FindPivot is that I hide a call to adapt the pivot metadata to a full pivot inside.
I think both are very viable, but I don't see a strong incentive to change it now that it's already written.
Thanks for the idea though.